service/src/main/java/bio/terra/tanagra/service/artifact/CohortService.java

```diff
@@ -125,6 +129,49 @@ public Cohort updateCohort(
     return cohortDao.getCohort(cohortId);
   }
 
+  /** Clone an existing cohort's revisions, including reviews and annotations */
+  public Cohort cloneCohort(
+      String studyId,
+      String cohortId,
+      String userEmail,
+      String destinationStudyId,
+      @Nullable String displayName,
+      @Nullable String description) {
+    Cohort original = getCohort(studyId, cohortId);
+
+    String newDisplayName = displayName;
+    if (newDisplayName == null && original.getDisplayName() != null) {
+      newDisplayName = original.getDisplayName();
+      if (studyId.equals(destinationStudyId)) {
+        newDisplayName = "Copy of: " + newDisplayName;
```

Comment:
Do we need to worry about the maximum length here?

Reply:
I searched for a max length but could not find anywhere on any artifact and hence did not add it.
Do we want to enforce it?

Reply:
Looks like it was https://github.com/DataBiosphere/tanagra/pull/945/files. We just don't want random failures because of it so we should truncate here if it's not happening elsewhere, which it doesn't look like.

Reply:
Thanks, added a check. However I will create a followup ticket to simplify it in openApi spec - https://verily.atlassian.net/browse/BENCH-4175